singlestoredb/management/job.py

```diff
@@ -0,0 +1,332 @@
+        self.max_allowed_execution_duration_in_minutes = max_allowed_execution_duration_in_minutes
+        self.notebook_path = notebook_path
+
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any]) -> 'ExecutionConfig':
+        """
+        Construct an ExecutionConfig from a dictionary of values.
+
+        Parameters
+        ----------
+        obj : dict
+            Dictionary of values
+
+        Returns
+        -------
+        :class:`ExecutionConfig`
+
+        """
+        out = cls(
+            create_snapshot=bool(obj.get('createSnapshot')),
+            max_allowed_execution_duration_in_minutes=int(obj.get('maxAllowedExecutionDurationInMinutes')),
+            notebook_path=obj.get('notebookPath'),
+        )
+
+        return out
+
+
+class Schedule(object):
+
+    execution_interval_in_minutes: int
+    start_at: datetime.datetime
+    mode: Optional[str]
+
+    def __init__(
+        self,
+        execution_interval_in_minutes: int,
+        start_at: datetime.datetime,
+        mode: Optional[str],
+    ):
+        self.execution_interval_in_minutes = execution_interval_in_minutes
+        self.start_at = start_at
+        self.mode = mode
+    
+    @classmethod
+    def from_dict(cls, obj: Dict[str, Any]) -> 'Schedule':
+        """
+        Construct a Schedule from a dictionary of values.
+
+        Parameters
+        ----------
+        obj : dict
+            Dictionary of values
+
+        Returns
+        -------
+        :class:`Schedule`
+
+        """
+        out = cls(
+            execution_interval_in_minutes=int(obj['executionIntervalInMinutes']),
+            start_at=to_datetime(obj['startAt']),
+            mode=obj.get('mode'),
+        )
+
+        return out
+
+
+class TargetConfig(object):
+    
+        database_name: Optional[str]
+        resume_target: Optional[bool]
+        target_id: Optional[str]
+        target_type: Optional[TargetType]
+    
+        def __init__(
+            self,
+            database_name: Optional[str],
+            resume_target: Optional[bool],
+            target_id: Optional[str],
+            target_type: Optional[TargetType]
+        ):
+            self.database_name = database_name
+            self.resume_target = resume_target
+            self.target_id = target_id
+            self.target_type = target_type
+    
+        @classmethod
+        def from_dict(cls, obj: Dict[str, Any]) -> 'TargetConfig':
+            """
+            Construct a TargetConfig from a dictionary of values.
+
+            Parameters
+            ----------
+            obj : dict
+                Dictionary of values
+
+            Returns
+            -------
+            :class:`TargetConfig`
+
+            """
+            out = cls(
+                database_name=obj.get('databaseName'),
+                resume_target=bool(obj.get('resumeTarget')),
+                target_id=obj.get('targetID'),
+                target_type=obj.get('targetType'),
```

Comment:
You'll want to map this to a `TargetType` enum like the `Status` change mentioned above.